### DISCUSSION TITLE
Otel B3 propagator is not matching traces between consumer record header and http request header

### DISCUSSION THREAD

**Discussion by mounikaVeeha** [open-telemetry/opentelemetry-java-instrumentation#5127] [Q&A]
I have a requirement to match the traces when a call goes from consumer to rest API.Here we are using b3 propogators . The traces in the record header are not matching with the traces in the rest API (HTTP Request).How to match the trace present in record header to trace in the request header as both headers have different traces (x-b3-traceId).

Also can we change/modify the agent generated Trace id (traceId generated using auto instrumentation ) with custom traceId .

**Comment by jkwatson** [comment]
Sounds like this is an auto-instrumentation question. I'll move it over to the appropriate place.
